prime-dotnet-webapi/Services/AgreementService.cs

```diff
@@ -278,6 +278,18 @@ public async Task<SignedAgreementDocument> AddSignedAgreementDocumentAsync(int a
             return signedAgreement;
         }
 
+        public async Task<IEnumerable<AgreementVersionViewModel>> GetLatestEnrolleeAgreementVersionsAsync()
+        {
+            var agreementVersionList = new List<AgreementVersion>();
+            foreach (var type in AgreementTypeExtensions.EnrolleeAgreementTypes())
+            {
+                agreementVersionList.Add(await FetchNewestAgreementVersionOfTypeAsync(type));
+            }
+            return agreementVersionList.AsQueryable()
+                .ProjectTo<AgreementVersionViewModel>(_mapper.ConfigurationProvider)
```

Comment:
You should almost never have to call `AsQueryable()` on something in this project. IQueryable more or less represents the application boundary that Entity Framework interfaces with to translate our C# code into SQL.  Automapper's `ProjectTo()` turns our ViewModel definitions into `Select()` statements that EF can then turn into SQL to retrieve our Models as ViewModels.

In this case, you have already materialized all the `AgreementVersion`s from the DB on line 286, so this collection already exists in memory. You should be able to just use

```C#
return _mapper.Map<IEnumerable<AgreementViewModel>>(agreementVersionList);
```
to do this.